scheduler/scheduler-client/src/main/java/org/ow2/proactive/scheduler/core/SchedulerStateImpl.java

```diff
@@ -60,13 +68,13 @@
 public final class SchedulerStateImpl<T extends JobState> implements SchedulerState<T> {
 
     /** Pending jobs */
-    private Set<T> pendingJobs = Collections.synchronizedSet(new LinkedHashSet<T>());
+    private Set<T> pendingJobs = new ConcurrentHashSet<>();
```

Comment:
by using ConcurrentHashSet we remove the insertion iteration order, which changes the orders of vectors returned by getPending, getRunning, etc.